src/components/InputWithSearch.tsx

```diff
@@ -0,0 +1,60 @@
+import { View, Text, TextInput, TouchableOpacity } from 'react-native';
+import { styles } from '../styles/inputWithSearch.styles';
+import { useState } from 'react';
+import { InputWithSearchProps } from '../types/input.types';
+
+export const InputWithSearch = ({
+  value,
+  placeholder,
+  options,
+  onSearch,
+  onChange,
+  iconRight,
+  iconClose,
+  iconOpen,
+  label,
+}: InputWithSearchProps) => {
+  const [isShowOptions, setIsShowOptions] = useState<boolean>(false);
+
+  const handleOnChange = (string: string) => {
+    onChange(string);
+  };
+
+  return (
+    <View style={[styles.container, isShowOptions ? styles.containerActive : null]}>
+      <TouchableOpacity onPress={() => setIsShowOptions(!isShowOptions)}>
+        <View style={styles.labelContainer}>
+          <Text style={[styles.labelText]}>{label ? label : 'Input text goes here'}</Text>
+          {isShowOptions ? iconClose : iconOpen}
+        </View>
+      </TouchableOpacity>
+
+      {isShowOptions ? (
```

Comment:
I think this prop is redundant, make this prop optional and set the default value as "true"